instrumentation/weblogic/src/main/java/com/splunk/opentelemetry/middleware/weblogic/WebLogicEntity.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.middleware.weblogic;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletRequest;
+
+/** Provides wrappers around WebLogic internal class instances. */
+public class WebLogicEntity {
+  private static final MethodHandle REQUEST_GET_CONTEXT;
+  private static final MethodHandle CONTEXT_GET_MBEAN;
+  private static final MethodHandle CONTEXT_GET_SERVER;
+  private static final MethodHandle SERVER_GET_MBEAN;
+  private static final MethodHandle MBEAN_GET_PARENT;
+  private static final MethodHandle MBEAN_GET_KEY;
+  private static final Class<?> MBEAN_CLASS;
+  private static final MethodHandle MBEAN_GET_ATTRIBUTE;
+
+  static {
+    MethodHandle requestGetContext;
+    MethodHandle contextGetMBean;
+    MethodHandle contextGetServer;
+    MethodHandle serverGetMBean;
+    MethodHandle mbeanGetParent;
+    MethodHandle mbeanGetKey;
+    MethodHandle mbeanGetAttribute;
+    Class<?> mbeanClass;
+
+    try {
+      requestGetContext =
+          MethodHandles.publicLookup()
+              .findVirtual(
+                  Class.forName("weblogic.servlet.internal.ServletRequestImpl"),
+                  "getContext",
+                  MethodType.methodType(
+                      Class.forName("weblogic.servlet.internal.WebAppServletContext")));
```

Comment:
Can you use compile stubs approach for this instead, as is done for liberty? https://github.com/signalfx/splunk-otel-java/tree/main/instrumentation/compile-stub

Reply:
One advantage of the current approach is that it's completely safe even in the absence of muzzle - we should preserve that feature and try-catch everything when converting to compile stub.

Reply:
This would add a lot of interfaces and classes with just one method in each, and zero in some - and this is if the interface hierarchy is changed to skip the intermediate ones we don't actually need, which is not so nice. An additional problem is that to have classes such as `ServletRequestImpl` stub also implement `ServletRequest`, we would need to include the servlet as a dependency for the stubs as well, which I am not sure is a good idea. So I'm not sure in this case the stubs would actually make it better, since even the below example uses inaccurate hierarchy, which is not very elegant.

![image](https://user-images.githubusercontent.com/3532037/104724919-0c3d5d80-573a-11eb-889c-3b813a632f4f.png)

Reply:
Let it stay as is then.